File: aoj/volume100/c/A10025.c
#include <math.h>
#include <stdio.h>

const double PI = 3.1415926535897932384626;

int main() {
	double a, b, c;
	double rad, deg;
	double area, perimeter, h;
	scanf("%lf %lf %lf", &a, &b, &deg);
	rad = deg * PI / 180.0;
	c = sqrt(pow(a, 2) + pow(b, 2) - 2*a*b*cos(rad));

	area      = a*b*sin(rad) / 2.0;
	perimeter = a + b + c;
	h         = 2*area / a;
	printf("%f\n", area);
	printf("%f\n", perimeter);
	printf("%f\n", h);
	return 0;
}

File: aoj/volume100/nodejs/A10025.js
function main(lines) {
  var nums = lines[0].split(/\s+/);
  var a   = parseFloat(nums[0]);
  var b   = parseFloat(nums[1]);
  var deg = parseFloat(nums[2]);
  var rad = deg * Math.PI / 180;
  var c   = Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2) - 2*a*b*Math.cos(rad));

  var area      = a*b*Math.sin(rad) / 2.0;
  var perimeter = a + b + c;
  var h         = 2*area / a;
  console.log([
    area.toFixed(6),
    perimeter.toFixed(6),
    h.toFixed(6),
  ].join(' '));
}

var chunks = [];
process.stdin.on('data', function(chunk) {
  chunks.push(chunk)
});
process.stdin.on('end', function() {
  var lines = chunks.join('').split('\n');
  main(lines);
});
